client/my-sites/site-settings/publishing-tools/post-by-voice.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button } from '@automattic/components';
+import { localizeUrl } from '@automattic/i18n-utils';
+import { ToggleControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import FormSettingExplanation from 'calypso/components/forms/form-setting-explanation';
+import SupportInfo from 'calypso/components/support-info';
+import { useSelector } from 'calypso/state';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+import { useGetPostByVoice } from './hooks/use-get-post-by-voice';
+import { useRegeneratePostByVoiceMutation } from './hooks/use-regenerate-post-by-voice-mutation';
+import { useSwitchPostByVoiceMutation } from './hooks/use-switch-post-by-voice-mutation';
+
+import './style.scss';
+
+type PostByVoiceSettingComponentProps = {
+	successNotice: typeof successNotice;
+	errorNotice: typeof errorNotice;
+};
+
+const PostByVoiceSettingComponent = ( {
+	successNotice,
+	errorNotice,
+}: PostByVoiceSettingComponentProps ) => {
+	const translate = useTranslate();
+	const selectedSiteId = useSelector( getSelectedSiteId );
+	const { data: postByVoiceSettings } = useGetPostByVoice( selectedSiteId );
+	const { mutate: switchPostByVoice, isPending: isPendingSwitch } =
+		useSwitchPostByVoiceMutation( selectedSiteId );
+	const { mutate: regeneratePostByVoice, isPending: isPendingRegenerate } =
+		useRegeneratePostByVoiceMutation( selectedSiteId );
+
+	const handleSwitch = ( checked: boolean ) => {
+		switchPostByVoice( checked, {
+			onSuccess: () => {
+				successNotice( translate( 'Changes saved successfully!' ) );
+			},
+			onError: () => {
+				errorNotice( translate( 'There was a problem saving your changes.' ) );
+			},
+		} );
+	};
+
+	const handleRegenerate = () => {
+		regeneratePostByVoice( undefined, {
+			onSuccess: () => {
+				successNotice( translate( 'Code regenerated successfully!' ) );
+			},
+			onError: () => {
+				errorNotice( translate( 'There was a problem regenerating the code.' ) );
+			},
+		} );
+	};
+
+	return (
+		<div>
+			<SupportInfo
+				text={ translate(
+					'Post by Voice is a way to publish audio posts to your blog with a phone call.'
+				) }
+				link={ localizeUrl( 'https://wordpress.com/support/post-by-voice/' ) }
+				privacyLink={ false }
+			/>
+
+			<ToggleControl
+				checked={ !! postByVoiceSettings?.isEnabled }
+				disabled={ isPendingSwitch || isPendingRegenerate }
+				label={ translate( 'Post by Voice' ) }
```

Comment:
```suggestion
				label={ translate( 'Post by Voice', {
					comment:
						'Post by Voice is the name of a blog setting that lets users post to their blogs by placing a phone call',
				} ) }
```
Let's add a translation comment. Always good practice